src/main/java/duke/core/Parser.java

```diff
@@ -0,0 +1,188 @@
+package duke.core;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeParseException;
+
+import duke.commands.AddCommand;
+import duke.commands.AddDeadlineCommand;
+import duke.commands.AddEventCommand;
+import duke.commands.AddTodoCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.InvalidCommandException;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.UnmarkCommand;
+import duke.tasks.TaskList;
+
+/**
+ * This class represents a parser that reads user input and converts it into Commands
+ */
+public class Parser {
+
+    /**
+     * Takes in user input and parses it into a Command that can be executed.
+     * TaskList-specific Commands will be executed on the given TaskList.
+     *
+     * @param userInput The String containing the user input.
+     * @param taskList  The TaskList that certain Commands will be executed on.
+     * @return The Command generated by parsing the user input.
+     * @throws InvalidCommandException if the user input is not a valid command.
+     */
+    public static Command parseUserInput(String userInput, TaskList taskList) throws InvalidCommandException {
+
+        //Check first word
+        String[] commandAndRemaining = userInput.strip().split(" ", 2);
+        String commandUpper = commandAndRemaining[0].toUpperCase();
+        Command cmd;
+        switch (commandUpper) {
```

Comment:
Good use of abstraction to parse the individual commands instead of lumping all of the logic in a code block! This improves readability and can help in tracing potential bugs by retracing the methods!